src/test/java/com/amazon/ion/conformance/expectations.kt

```diff
@@ -53,6 +54,7 @@ fun TestCaseSupport.assertSignals(sexp: SeqElement, r: IonReader) {
 private fun IonReader.walk(): List<String> {
     val events = mutableListOf<String>()
     fun recordEvent(eventType: String = type.toString(), value: Any? = "") {
+        if (events.size > 10_000_000) fail("Ion stream does not appear to terminate.")
```

Comment:
🗺️ Useful safeguard for when you introduce an infinite loop, or you start writing a conformance DSL test case consisting of a "Billion Laughs" attack.